src/overlay.py

```diff
@@ -27,20 +30,91 @@ def display(self, time):
+
+            self.rect = pygame.Rect(self.left, self.top, width, height)
+            self.display = self.display_digital
+
+    def display_analog(self):
+
+        # get time
+        time = self.game_time.get_time()
+        hour = time[0] % 12
+        minute = time[1]
+
+        # frame
+        pygame.draw.rect(self.display_surface, 'White', self.rect, 0, 10)
+        pygame.draw.rect(self.display_surface, 'Black', self.rect, 5, 10)
+        
+        # hands position
+        hour_hand_angle = 2 * pi * (hour * 60 + minute) / (12 * 60) - pi / 2
+        minute_hand_angle = 2 * pi * minute / 60 - pi / 2
+        hour_vector = self.center + 0.75 * self.hand_lenght * pygame.math.Vector2(cos(hour_hand_angle), sin(hour_hand_angle)) 
+        minute_vector = self.center + self.hand_lenght * pygame.math.Vector2(cos(minute_hand_angle), sin(minute_hand_angle))
+
+        # draw hands
+        pygame.draw.line(self.display_surface, 'Black', self.center, minute_vector, 5)
+        pygame.draw.line(self.display_surface, 'Black', self.center, hour_vector, 5)
+        pygame.draw.circle(self.display_surface, 'Black', self.center, 4)
+
+    def display_digital(self):
+
+        # get time
+        time = self.game_time.get_time()
+        hour = time[0]
+        minute = time[1]
+
+        if hour < 10:
             # if hours are less than 10, add a 0 to stay in the hh:mm format
-            hours = f"0{time[0]}"
+            hour_text = f'0{hour}'
         else:
-            hours = f"{time[0]}"
+            hour_text = f'{hour}'
 
-        if (time[1] < 10):
+        if (minute < 10):
             # if minutes are less than 10, add a 0 to stay in the hh:mm format
-            minutes = f"0{time[1]}"
+            minute_text = f'0{minute}'
         else:
-            minutes = f"{time[1]}"
+            minute_text = f'{minute}'
+
+        # rects and surfs
+        pady = 2
+
+        colon_surf = self.font.render(':', False, 'Black')
+        colon_rect = colon_surf.get_frect(center = (self.rect.centerx, self.rect.centery + pady))
+
+        hour_surf = self.font.render(str(hour_text), False, 'Black')
+        hour_rect = hour_surf.get_frect(midright = (self.rect.centerx - colon_rect.width, self.rect.centery + pady))
+
+        minute_surf = self.font.render(str(minute_text), False, 'Black')
+        minute_rect = minute_surf.get_frect(midleft = (self.rect.centerx + colon_rect.width, self.rect.centery + pady))
 
-        text_surface = font.render(
-            f"{hours}:{minutes}", False, (255, 255, 255))
-        self.display_surface.blit(text_surface, (10, 10))
+        # display
+        pygame.draw.rect(self.display_surface, 'White', self.rect, 0, 4)
+        pygame.draw.rect(self.display_surface, 'Black', self.rect, 4, 4)
+        self.display_surface.blit(colon_surf, colon_rect)
+        self.display_surface.blit(hour_surf, hour_rect)
+        self.display_surface.blit(minute_surf, minute_rect)
```

Comment:
All this could be replaced by a single font.render and blit call. You might want to use f-strings instead when rendering the text

Reply:
I know I can use only one f-string, but I want to fix the position of ':', this way I obtained a way nicer clock were the text doesn't move based on the length of the text (digits have different pixels-width)

Reply:
This could be fixed by editing the font itself, but I'm not sure we have the legal rights to do so